restaurants/views.py

```diff
@@ -141,10 +142,15 @@ class ReviewCreateView(LoginRequiredMixin, CreateView):
     template_name = 'restaurants/review_form.html'
 
     def dispatch(self, request, *args, **kwargs):
-        self.restaurant = get_object_or_404(Restaurant, pk=self.kwargs['id'])
+        self.restaurant = get_object_or_404(Restaurant, pk=self.kwargs['restaurant_id'])
         return super().dispatch(request, *args, **kwargs)
 
     def form_valid(self, form): 
+        existing_review = Review.objects.filter(user=self.request.user, restaurant=self.restaurant).first()
+        if existing_review:
+            messages.warning(self.request, "You have already reviewed this restaurant.")
+            return redirect('restaurant-detail', pk=self.restaurant.pk)
```

Comment:
It is fine that are you handling this in view, but the "Write Review" button also should not be shown to me if I have already given review. Instead "Edit review" and "Delete Review" button should show.